File: base/src/main/java/mf/asciitext/FontPickerAdapter.java
package mf.asciitext;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import mf.asciitext.fonts.AppFont;

public class FontPickerAdapter extends RecyclerView.Adapter<FontPickerAdapter.ItemViewHolder> {

    private final List<AppFont> dataSource;

    private final OnItemClickListener mItemClickListener;

    private final int activeItemType = 1;

    private int selectedFont;

    interface OnItemClickListener {
        void onItemClick(AppFont item, int index);
    }

    public FontPickerAdapter(List<AppFont> data, OnItemClickListener onItemClick) {
        dataSource = data;
        mItemClickListener = onItemClick;
    }

    public void setSelectedFont(int index) {
        selectedFont = index;
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        int layoutRes = (viewType ==  activeItemType ? R.layout.item_font_selected : R.layout.item_font);
        return new ItemViewHolder(LayoutInflater.from(parent.getContext()).inflate(layoutRes, parent, false));
    }

    @Override
    public int getItemViewType(int position) {
        return (position == selectedFont && selectedFont>=0) ? activeItemType : -1;
    }

    @Override
    public void onBindViewHolder(@NonNull final FontPickerAdapter.ItemViewHolder holder, int position) {
        final AppFont value = dataSource.get(position);
        holder.mTextView.setText(value.GetStyledName());
    }

    @Override
    public int getItemCount() {
        return dataSource == null ? 0 : dataSource.size();
    }

    @SuppressWarnings("unused")
    public class ItemViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        private final TextView mTextView;

        ItemViewHolder(View v) {
            super(v);
            mTextView = v.findViewById(R.id.text_value);
            View container = v.findViewById(R.id.text_container);

            /* when user clicks on a recyclerView item */
            container.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            try {
                final int index = getAdapterPosition();
                AppFont item = dataSource.get(index);
                mItemClickListener.onItemClick(item, index);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }
}

File: base/src/main/java/mf/asciitext/FontPickerAdapter.kt
package mf.asciitext

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import mf.asciitext.FontPickerAdapter.ItemViewHolder
import mf.asciitext.fonts.AppFont

class FontPickerAdapter(
    private val dataSource: List<AppFont>?,
    private val mItemClickListener: OnItemClickListener
) : RecyclerView.Adapter<ItemViewHolder>() {
    private val activeItemType = 1
    private var selectedFont = 0

    interface OnItemClickListener {
        fun onItemClick(item: AppFont?, index: Int)
    }

    fun setSelectedFont(index: Int) {
        selectedFont = index
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val layoutRes =
            if (viewType == activeItemType) R.layout.item_font_selected else R.layout.item_font
        return ItemViewHolder(
            LayoutInflater.from(parent.context).inflate(layoutRes, parent, false)
        )
    }

    override fun getItemViewType(position: Int): Int {
        return if (position == selectedFont && selectedFont >= 0) activeItemType else -1
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        val value = dataSource!![position]
        holder.mTextView.text = value.styledName
    }

    override fun getItemCount(): Int {
        return dataSource?.size ?: 0
    }

    inner class ItemViewHolder internal constructor(v: View) :
        RecyclerView.ViewHolder(v), View.OnClickListener {
        val mTextView: TextView = v.findViewById(R.id.text_value)
        override fun onClick(v: View) {
            try {
                val index = adapterPosition
                val item = dataSource!![index]
                mItemClickListener.onItemClick(item, index)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }

        init {
            val container =
                v.findViewById<View>(R.id.text_container)

            container.setOnClickListener(this)
        }
    }
}